cmd/wstail/forwarders.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/gravitational/logging-app/lib/forwarders"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/gravitational/trace"
+)
+
+// updateForwarders updates log forwarder configuration and reloads the logging service.
+// It receives new forwarder configuration, updates the configuration and restarts the rsyslog daemon
+// to force it to reload the configuration
+func updateForwarders(w http.ResponseWriter, r *http.Request) (err error) {
+	if r.Method != "PUT" {
+		return trace.BadParameter("invalid HTTP method: %v", r.Method)
+	}
+	var forwarders []forwarders.Forwarder
+	if err = readJSON(r, &forwarders); err != nil {
+		return trace.Wrap(err)
+	}
+
+	// Remove previous forwarder configuration files
+	if err = filepath.Walk(rsyslogConfigDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		if path == rsyslogConfigDir {
+			return nil
+		}
+		if info.IsDir() {
+			return filepath.SkipDir
+		}
+		log.Infof("removing %v", path)
+		return os.Remove(path)
+	}); err != nil {
+		log.Warningf("failed to delete forwarder configuration files: %v", err)
+	}
+
+	// Write new forwarder configuration
+	for _, forwarder := range forwarders {
+		f, err := os.Create(path(forwarder))
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		_, err = f.WriteString(config(forwarder))
+		f.Close()
+		if err != nil {
+			return trace.Wrap(err)
+		}
+	}
+
+	// Reload rsyslogd
+	if out, err := exec.Command("/etc/init.d/rsyslog", "restart").CombinedOutput(); err != nil {
```

Comment:
I suspect the files are there, but yeah, won't work. Might be able to just `hup` the rsyslog pid though.